examples/Model/User.php

```diff
@@ -45,8 +45,12 @@ function changeUserName(callable $stateResolver, ChangeUserName $command): array
 
 const apply = '\Prooph\MicroExample\Model\User\apply';
 
-function apply(array $state, Message ...$events): array
+function apply($state, Message ...$events): array
```

Comment:
This function requires `$state` to be an `array` or `null` and would raise an exception otherwise. 
So I think `?array $state` is the better type hint here.

Reply:
array_merge would not work with an object. Also if $events is an empty array, $state would be returned untouched. The return type hint is `array` and could therefore not be an `object` without raising an exception.